src/lavinmq/amqp/queue/stream_queue_message_store.cr

```diff
@@ -126,12 +126,23 @@ module LavinMQ::AMQP
           sp = SegmentPosition.new(consumer.segment, consumer.pos, msg.bytesize.to_u32)
           consumer.pos += sp.bytesize
           consumer.offset += 1
+          if consumer_filter = consumer.filter
+            return unless matching?(msg.properties.headers, consumer_filter, consumer.match_unfiltered?)
+          end
```

Comment:
I think it would be nicer to move this to the consumer. Then the consumer doesn't need to make `filter` and `match_unfiltered` public. 
```suggestion
          return unless consumer.filter_match?(msg.properties.headers)
```